src/views/Predictions/components/RoundCard/OpenRoundCard.tsx

```diff
@@ -74,13 +75,21 @@ const OpenRoundCard: React.FC<OpenRoundCardProps> = ({
     }))
   }
 
-  const handleSuccess = async () => {
+  const handleSuccess = async (decimalValue: BigNumber, hash: string) => {
     const positionDisplay = position === BetPosition.BULL ? 'UP' : 'DOWN'
 
-    setState((prevState) => ({
-      ...prevState,
-      hasEnteredPosition: true,
-    }))
+    // Optimistically set the user bet so we see the entered position immediately.
+    dispatch(
+      markPositionAsEntered({
+        account,
+        roundId: round.id,
+        partialBet: {
+          hash,
+          position,
+          amount: getBnbAmount(decimalValue).toNumber(),
+        },
+      }),
+    )
```

Comment:
Here is where we set the partial bet immediately so the UI updates. I tried fetching the bet from the graph but it is not updated quick enough.